src/TwigComponent/src/Twig/PropsNode.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent\Twig;
+
+use Twig\Compiler;
+use Twig\Node\Node;
+
+/**
+ * @author Matheo Daninos <matheo.daninos@gmail.com>
+ *
+ * @internal
+ */
+class PropsNode extends Node
+{
+    public function __construct(array $propsNames, array $values, $lineno = 0, string $tag = null)
+    {
+        parent::__construct($values, ['names' => $propsNames], $lineno, $tag);
+    }
+
+    public function compile(Compiler $compiler): void
+    {
+        foreach ($this->getAttribute('names') as $name) {
+            $compiler
+                ->addDebugInfo($this)
+                ->write('if (!isset($context[\''.$name.'\'])) {')
+            ;
+
+            if (!$this->hasNode($name)) {
+                $compiler
+                    ->write('throw new \Exception("'.$name.' should be defined for component '.$this->getTemplateName().'");')
```

Comment:
And here to make something more helpful

Reply:
> The prop "$name" defined in the {% props %} tag was not passed when rendering the component.

Also, I'm not sure which type of `\Exception` should be thrown - but maybe you can check out some similar code.